proxy/http/HttpSM.cc

```diff
@@ -7029,13 +7023,13 @@ HttpSM::dump_state_on_assert()
 {
   Error("[%" PRId64 "] ------- begin http state dump -------", sm_id);
 
-  int hist_size = this->history_pos;
-  if (this->history_pos > HISTORY_SIZE) {
-    hist_size = HISTORY_SIZE;
-    Error("   History Wrap around. history_pos: %d", this->history_pos);
+  unsigned int hist_size = history.history_pos;
+  if (hist_size > history.history_length) {
+    hist_size = history.history_length;
+    Error("   History Wrap around. history_pos: %d", history.history_pos);
```

Comment:
If the history actually overflowed, then you would have clobbered memory in `push_back()`. How about in `push_back()` allowing `history_pos` to increment but only writing the history entry if it is within bounds. Then it is easy to add a `overflowed()` member function by comparing `history_pos` to `Count`. Should be easy to ass a unit test for this too.